src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1969,6 +1970,30 @@ private void HandleVariableDeclaration(IVariableDeclarationOperation operation)
             {
                 ILocalSymbol localSymbol = declarator.Symbol;
 
+                // We skip constants in the control flow graph, as they're not actually involved in any control flow.
+                if (localSymbol.IsConst)
+                {
+                    continue;
+                }
+
+                // If the local is a static (possible in VB), then we create a semaphore for conditional execution of the initializer.
+                BasicBlock afterInitialization = null;
+                if (localSymbol.IsStatic && (declarator.Initializer != null || operation.Initializer != null))
+                {
+                    afterInitialization = new BasicBlock(BasicBlockKind.Block);
+                    LinkBlocks(CurrentBasicBlock, afterInitialization);
+
+                    ITypeSymbol booleanType = ((Operation)operation).SemanticModel.Compilation.GetSpecialType(SpecialType.System_Boolean);
+                    var initializationSemaphore = new StaticLocalInitializationSemaphoreOperation(localSymbol, localSymbol.DeclaringSyntaxReferences.Single().GetSyntax(), booleanType, constantValue: null, isImplicit: true);
```

Comment:
>localSymbol.DeclaringSyntaxReferences.Single().GetSyntax() [](start = 111, length = 58)

I don't think it is a good idea to make an assumption that we can always get syntax like that, we should simply use syntax from the IOperation nodes at hand.

Reply:
Fixed.

---
In reply to: [174859099](https://github.com/dotnet/roslyn/pull/25458#discussion_r174859099) [](ancestors = 174859099)